server/src/main/java/org/apache/iotdb/db/engine/tsfilemanagement/level/LevelTsFileManagement.java

```diff
@@ -0,0 +1,508 @@
+import org.apache.iotdb.db.engine.tsfilemanagement.utils.HotCompactionLogger;
+import org.apache.iotdb.db.engine.tsfilemanagement.utils.HotCompactionUtils;
+import org.apache.iotdb.db.query.control.FileReaderManager;
+import org.apache.iotdb.tsfile.file.metadata.ChunkMetadata;
+import org.apache.iotdb.tsfile.fileSystem.FSFactoryProducer;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.write.schema.MeasurementSchema;
+import org.apache.iotdb.tsfile.write.writer.RestorableTsFileIOWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LevelTsFileManagement extends TsFileManagement {
+
+  private static final Logger logger = LoggerFactory.getLogger(LevelTsFileManagement.class);
+  private int maxLevelNum = IoTDBDescriptor.getInstance().getConfig().getMaxLevelNum();
+  private final List<List<TsFileResource>> sequenceTsFileResources = new CopyOnWriteArrayList<>();
+  private final List<List<TsFileResource>> unSequenceTsFileResources = new CopyOnWriteArrayList<>();
+  private final List<List<TsFileResource>> forkedSequenceTsFileResources = new ArrayList<>();
+  private final List<List<TsFileResource>> forkedUnSequenceTsFileResources = new ArrayList<>();
+
+  public LevelTsFileManagement(String storageGroupName, String storageGroupDir) {
+    super(storageGroupName, storageGroupDir);
+    clear();
+  }
+
+  private void deleteLevelFiles(List<TsFileResource> vmMergeTsFiles) {
+    logger.debug("{} [hot compaction] merge starts to delete file", storageGroupName);
+    for (TsFileResource vmMergeTsFile : vmMergeTsFiles) {
+      deleteLevelFile(vmMergeTsFile);
+    }
+    for (int i = 0; i < maxLevelNum; i++) {
+      sequenceTsFileResources.get(i).removeAll(vmMergeTsFiles);
+      unSequenceTsFileResources.get(i).removeAll(vmMergeTsFiles);
+    }
+  }
+
+  private static void deleteLevelFile(TsFileResource seqFile) {
+    seqFile.writeLock();
+    try {
+      ChunkMetadataCache.getInstance().remove(seqFile);
+      FileReaderManager.getInstance().closeFileAndRemoveReader(seqFile.getTsFilePath());
+      seqFile.setDeleted(true);
+      if (seqFile.getTsFile().exists()) {
+        Files.delete(seqFile.getTsFile().toPath());
+      }
+    } catch (Exception e) {
+      logger.error(e.getMessage(), e);
+    } finally {
+      seqFile.writeUnlock();
+    }
+  }
+
+  private void flushAllFilesToLastLevel(List<List<TsFileResource>> currMergeFiles,
+      HotCompactionLogger hotCompactionLogger, boolean sequence,
+      ReadWriteLock hotCompactionMergeLock) throws IOException {
+    TsFileResource sourceFile = currMergeFiles.get(0).get(0);
+    File newTargetFile = createNewTsFileName(sourceFile.getTsFile(), maxLevelNum - 1);
+    TsFileResource targetResource = new TsFileResource(newTargetFile);
+    HotCompactionUtils.merge(targetResource, getTsFileList(sequence),
+        storageGroupName, hotCompactionLogger, new HashSet<>(), sequence);
```

Comment:
It seems that we should use currMergeFiles instead of `getTsFileList(sequence)`.